### ISSUE TITLE
Support React 17.0.2

### ISSUE THREAD

**Issue by lowecg** [reagent-project/reagent#536]
Is it possible to use React version 17.0.2? My builds are outputting the following warnings:

```
NPM dependency "react" has installed version "17.0.2"
"17.0.1" was required by jar:file:/Users/lowecg/.m2/repository/reagent/reagent/1.0.0/reagent-1.0.0.jar!/deps.cljs
NPM dependency "react-dom" has installed version "17.0.2"
"17.0.1" was required by jar:file:/Users/lowecg/.m2/repository/reagent/reagent/1.0.0/reagent-1.0.0.jar!/deps.cljs
```

The React release for 17.0.2 seems worthwhile since it addresses a "SharedArrayBuffer cross-origin isolation" warning in Chrome.

https://github.com/facebook/react/blob/master/CHANGELOG.md#1702-march-22-2021

Forgive a potentially naive question - does `deps.cljs` support an NPM-style semver wildcarding (^ = latest minor, ~ = latest patch). If so, would latest patch be worth considering?

**Comment by Deraen** [comment]
I don't think deps.cljs supports wildcards, but I'm not sure.

Anyway, those warnings are from Shadow-cljs?

Reagent itself should be fully compatible with this React, seeing that this is minor release. Will be fine to update default React in Reagent, but there won't be Reagent release for every React update to just update deps. You could just ignore those warnings when you select the React version yourself, or find a way to disable these warnings in the build tool you use.

**Comment by lowecg** [comment]
Understood and thank you for the response.

Those warnings are from Shadow-cljs, yes. Sorry, I forgot to mention that!

I couldn't see a way to disable these particular warnings, but I did end up disabling NPM dependencies from Shadow-cljs entirely which certainly does remove the warning.

I found a similar issue on the shadow-cljs project. According to the following comment, the CLJSJS dependency isn't used anyway so turning off NPM dependencies is the only course.

https://github.com/thheller/shadow-cljs/issues/800#issuecomment-725716087

> You can set :npm-deps {:install false} in shadow-cljs.edn at the top level.
